src/easylink/pipeline.py

```diff
@@ -14,60 +25,121 @@
 
 
 class Pipeline:
-    """Abstraction to handle pipeline specification and execution."""
+    """A convenience class for validations and Snakefile generation.
```

Comment:
I think this is right, but as Jim pointed out, it kind of exposes that this class isn't super focused. Maybe validations should be somewhere separate and this just becomes a SnakefileWriter or something...